### PR TITLE
Remove legacy formats

### PR DESCRIPTION
**Pull request by akhenry**
Fixes https://github.com/nasa/openmct/issues/4530

Note that this PR removes support for legacy format registration completely. See deprecation notes on the related issue.

### All Submissions:

* [X] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [X] Changes address original issue?
* [X] Unit tests included and/or updated with changes?
* [X] Command line build passes?
* [X] Has this been smoke tested?
* [X] Testing instructions included in associated issue?

### CODE REVIEW

File: src/plugins/utcTimeSystem/UTCTimeFormat.js [hunk 1]
```diff
@@ -20,75 +20,62 @@
...
      * @param {number} value The value to format.
-     * @param {string} formatString The string format to format. Default "YYYY-MM-DD HH:mm:ss.SSS" + "Z"
-     * @returns {string} the formatted date(s) according to the proper parameter of formatString or the default value of "YYYY-MM-DD HH:mm:ss.SSS" + "Z".
-     * If multiple values were requested, then an array of
+     * @returns {string} the formatted date(s). If multiple values were requested, then an array of
      * formatted values will be returned. Where a value could not be formatted, `undefined` will be returned at its position
      * in the array.
      */
-    UTCTimeFormat.prototype.format = function (value, formatString) {
+    format(value) {
```
Comment (shefalijoshi) [inline src/plugins/utcTimeSystem/UTCTimeFormat.js:59]: Does this mean we're not allowing user's to specify the `formatString` anymore? Looks like a breaking change here.

**Review by shefalijoshi — COMMENTED** [review]
One comment, but the rest looks good.
Nice tests too!

File: src/plugins/utcTimeSystem/UTCTimeFormat.js (hunk 1 above)
Comment (akhenry) [reply src/plugins/utcTimeSystem/UTCTimeFormat.js:59]: Stand by, I think something odd is going on here. It looks like another PR was merged in the mean time that also migrates the UTCTimeFormat to a plugin :)

**Comment by akhenry** [comment]
Ok, wow that was a mess (entirely of my own making - I should have been aware of it). [Another PR was merged in the mean time](https://github.com/nasa/openmct/pull/4412) that covers some of the same ground, and I had to merge the two approaches. We're good to go though.

File: src/plugins/displayLayout/CustomStringFormatterSpec.js [hunk 2]
```diff
@@ -44,12 +44,14 @@ describe('CustomStringFormatter', function () {
...
-        CUSTOM_FORMATS.forEach(openmct.telemetry.addFormat.bind({openmct}));
+        CUSTOM_FORMATS.forEach((formatter) => {
+            openmct.telemetry.addFormat(formatter);
+        });
         openmct.on('start', done);
         openmct.startHeadless();
 
-        spyOn(openmct.telemetry, 'getFormatter');
-        openmct.telemetry.getFormatter.and.callFake((key) => CUSTOM_FORMATS.find(d => d.key === key));
+        /*spyOn(openmct.telemetry, 'getFormatter');
```
Comment (shefalijoshi) [inline src/plugins/displayLayout/CustomStringFormatterSpec.js:53]: If we don't need this, lets remove it.

**Review by shefalijoshi — APPROVED** [review]
Looks good!

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate unit tests included?
* [x] Code style and in-line documentation are appropriate?
* [x] Commit messages meet standards?
* [x] Has associated issue been labelled `unverified`? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled `bug`? (only applicable if this PR is for a bug fix)
* [ ] List of Acceptance Tests Performed.

File: src/plugins/displayLayout/CustomStringFormatterSpec.js (hunk 2 above)
Comment (akhenry) [reply src/plugins/displayLayout/CustomStringFormatterSpec.js:53]: Good catch, thanks!

File: src/plugins/utcTimeSystem/UTCTimeFormat.js (hunk 1 above)
Comment (akhenry) [reply src/plugins/utcTimeSystem/UTCTimeFormat.js]: Fixed

File: src/plugins/displayLayout/CustomStringFormatterSpec.js (hunk 2 above)
Comment (akhenry) [reply src/plugins/displayLayout/CustomStringFormatterSpec.js]: Fixed

**Review by shefalijoshi — APPROVED** [review]

**Comment by davetsay** [comment]
@akhenry , @shefalijoshi , I made the changes in VISTA. Everything seems to be working okay.

I do notice the time selector is breaking in realtime mode. I don't _think_ its related though.

**Comment by davetsay** [comment]
@akhenry , @shefalijoshi , okay it was related. I resolved the realtime inputs not working in VISTA by installing the new `UTCTimeSystem` plugin, which contains the `DurationFormat`.

**Pull request merged**